core/src/validate/validator.rs

```diff
@@ -70,41 +82,52 @@ fn validate_foreign_strings(
 
     let apos_result = apostrophe::validate(&foreign_strings);
     let fs_result = format_string::validate(&mut default_parsed_data, &mut foreign_strings);
+    let ms_result = missing_strings::validate(&mut default_strings, &mut foreign_strings);
 
-    let invalid = if apos_result.is_err() && fs_result.is_err() {
-        Some(InvalidStringsFile {
-            file_path: foreign_strings_file_path.clone(),
-            apostrophe_error: Some(apos_result.unwrap_err()),
-            format_string_error: Some(fs_result.unwrap_err()),
-        })
-    } else if apos_result.is_err() {
-        Some(InvalidStringsFile {
-            file_path: foreign_strings_file_path.clone(),
-            apostrophe_error: Some(apos_result.unwrap_err()),
-            format_string_error: None,
-        })
-    } else if fs_result.is_err() {
-        Some(InvalidStringsFile {
-            file_path: foreign_strings_file_path.clone(),
-            apostrophe_error: None,
-            format_string_error: Some(fs_result.unwrap_err()),
-        })
-    } else {
-        None
-    };
+    let mut potential_invalid_file = InvalidStringsFile::new(foreign_strings_file_path.clone());
+
+    if let Err(apos_error) = apos_result {
+        potential_invalid_file.apostrophe_error = Some(apos_error);
+    }
+
+    if let Err(fs_error) = fs_result {
+        potential_invalid_file.format_string_error = Some(fs_error);
+    }
+
+    if fail_on_unlocalized {
+        if let Err(ms_error) = ms_result {
+            potential_invalid_file.missing_strings_error = Some(ms_error);
+        }
+    }
 
-    if let Some(invalid_file) = invalid {
-        invalid_strings_files.push(invalid_file)
+    if potential_invalid_file.has_errors() {
+        invalid_strings_files.push(potential_invalid_file)
     } else {
         path_of_validated_files.push(foreign_strings_file_path)
     }
 }
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, Default, PartialEq)]
 pub struct InvalidStringsFile {
     pub file_path: String,
     pub apostrophe_error: Option<apostrophe::InvalidStrings>,
     pub format_string_error: Option<format_string::Mismatches>,
+    pub missing_strings_error: Option<missing_strings::MissingStrings>,
+}
+
+impl InvalidStringsFile {
+    fn new(file_path: String) -> InvalidStringsFile {
+        InvalidStringsFile {
+            file_path,
+            ..Default::default()
+        }
+    }
+
+    fn has_errors(&self) -> bool {
+        self.apostrophe_error.is_some()
+            || self.format_string_error.is_some()
+            || self.missing_strings_error.is_some()
+    }
 }
```

Comment:
can this be in its own file?

Reply:
It could be. Reason I kept it here: other files in the same modules also have structs with which they express their result & those structs just live inside those modules. So thought it would be uniform to do it for this too...